vminfo_parser/vminfo_parser.py

```diff
@@ -94,6 +94,54 @@ def add_extra_columns(self: t.Self) -> None:
         else:
             print("All columns already exist")
 
+    def add_site_specific_columns(self: t.Self) -> pd.DataFrame:
```

Comment:
given the changed way this works,  is this name appropriate?   Also,  given that its no longer modifying the dataframe, but using it to calculate  some data, and return a new frame,  which it be in analyzer instead of vmdata?

Reply:
changed to `create_site_specific_dataframe`